crates/protocol/src/block.rs

```diff
@@ -62,7 +66,23 @@ impl L2BlockInfo {
         Self { block_info, l1_origin, seq_num }
     }
 }
-
+/// OP Stack full block.
+///
+/// Withdrawals can be optionally included at the end of the RLP encoded message.
+///
+/// Taken from [reth-primitives](https://github.com/paradigmxyz/reth)
+#[derive(Debug, Clone, PartialEq, Eq, Default, RlpEncodable, RlpDecodable)]
+#[rlp(trailing)]
+pub struct OpBlock {
+    /// Block header.
+    pub header: Header,
+    /// Transactions in this block.
+    pub body: Vec<OpTxEnvelope>,
+    /// Ommers/uncles header.
+    pub ommers: Vec<Header>,
+    /// Block withdrawals.
+    pub withdrawals: Option<Vec<Withdrawal>>,
+}
```

Comment:
We want to use https://github.com/alloy-rs/alloy/blob/main/crates/rpc-types-eth/src/block.rs#L20 instead of duplicating the block definition here. We can then alias `Block<OpTxEnvelope>` as `OpBlock`. The issue right now is `alloy-rpc-types-eth` doesn't support `no_std` _and_ requires serde by default. I have a pr open for this here: https://github.com/alloy-rs/alloy/pull/1252

Reply:
> We want to use https://github.com/alloy-rs/alloy/blob/main/crates/rpc-types-eth/src/block.rs#L20 instead of duplicating the block definition here. We can then alias `Block<OpTxEnvelope>` as `OpBlock`. The issue right now is `alloy-rpc-types-eth` doesn't support `no_std` _and_ requires serde by default. I have a pr open for this here: [alloy-rs/alloy#1252](https://github.com/alloy-rs/alloy/pull/1252)

thank you for your fast reply, i'll wait your pr get merged to use Block type then!